src/Components/Web.JS/src/Boot.Web.ts

```diff
@@ -30,6 +31,14 @@ function boot(options?: Partial<WebStartOptions>) : Promise<void> {
   started = true;
 
   Blazor._internal.loadWebAssemblyQuicklyTimeout = 3000;
+  // Definedh ere to avoid inadvertently imported enhanced navigation
+  // related APIs in WebAssembly or Blazor Server contexts.
+  Blazor._internal.hotReloadApplied = () => {
+    if (hasProgrammaticEnhancedNavigationHandler())
+    {
+      performProgrammaticEnhancedNavigation(location.href, false);
```

Comment:
I think we want the `replace` argument to be `true` here because we don't want these refreshes to get pushed to the history stack.
```suggestion
      performProgrammaticEnhancedNavigation(location.href, true);
```